External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -1008,13 +1008,16 @@ static private void HandleStructureCompletion(ScintillaControl Sci)
                 int line = Sci.LineFromPosition(position);
                 if (line == 0)
                     return;
-                string txt = Sci.GetLine(line-1).TrimEnd();
+                string txt = Sci.GetLine(line - 1).TrimEnd();
                 int style = Sci.BaseStyleAt(position);
 
+                // move closing brace to its own line and fix indentation
                 if (Sci.CurrentChar == '}')
                 {
-                    Sci.DeleteForward();
-                    AutoCloseBrace(Sci, line);
+                    var openingBrace = Sci.SafeBraceMatch(position);
+                    var openLine = openingBrace >= 0 ? Sci.LineFromPosition(openingBrace) : line - 1;
+                    Sci.InsertText(Sci.CurrentPos, LineEndDetector.GetNewLineMarker(Sci.EOLMode));
```

Comment:
Good reminder, but `Sci.NewLine` is going to change the cursor position as well. I'll give it a try though.